### PR TITLE
Update AES-based header protection algorithm to use AES-ECB

### PR DESCRIPTION
**Pull request by ueno**
Until draft-16, the header protection mask was calculated using AES-CTR, while it has changed to use AES-ECB in draft-17. Although I suspect this is kept for interoperability reasons, it would be nice to use ECB as some crypto libs don't support CTR (single block ECB can be emulated with CBC, which is available everywhere).

### CODE REVIEW

**Comment by tatsuhiro-t** [comment]
Thank you for PR.  It looks like you just changed implementation detail of the particular crypto library (openssl) and it supports AES-CTR.  If "some crypto libs" mean libraries other than openssl, then I think we just say in documentation that AES-ECB can be used here if AES-CTR is not available.

**Comment by ueno** [comment]
Sorry, I should have added a bit more context. Let me rephrase: I'm trying to write a backend using GnuTLS, and realized that our AES-ECB based implementation doesn't interoperate with the stock `examples/server` compiled with the OpenSSL backend, because AES-CTR and AES-ECB are completely different algorithms. I'm not sure if it's sufficient to just document the difference.

**Comment by ueno** [comment]
Sorry, it turned out to be completely my misunderstanding: indeed both calculate the same value, and the bug was in my backend code. I apologize for the noise.
